src/Storefront/Controller/AuthController.php

```diff
@@ -86,6 +86,7 @@ public function loginPage(Request $request, RequestDataBag $data, SalesChannelCo
             'waitTime' => $request->get('waitTime'),
             'errorSnippet' => $request->get('errorSnippet'),
             'data' => $data,
+            'statusCode' => (bool) $request->get('loginError') ? 401 : 200
```

Comment:
@tobiasberge & @ssltg any thoughts on this? Do we wanna use non-200 status code for properly rendered twig?

Reply:
Also based on the linked SO post from above, the semantically correct status code probably would be 400 and not 401

Reply:
1) a response status is not linked to whether a twig was properly rendered; a 404 page is also properly rendered.
2) the linked SO post is about an OAuth flow. The login page is a HTML form/page, which is the underlying mechanism, so the proper response is imo 401 since the request was totally fine (so not 400 Bad Request).
2b) the same point goes for "_The 400 error is spec compliant_", as in that's the spec for OAuth authorization.

Reply:
It is a bit twisted. The original request was the Post to the login route and if the login was not successfulll, we "internally" redirect to the login page with the error message. 
In my opinion the storefront should never send content with a post request (that is not an Ajax request) to the browser. So I would rather change this to make a proper answer of the login post with status 401 that redirects with a header to the loginpage via GET and status 200

Reply:
I think I agree @ssltg. The redirect would happen in the handling of the response of the Ajax request in the JS, right?

And just a little more context for clarity: we now can't distinguish failed from successful request, so a brute force attack can't be detected (by fail2ban for example), or at least not based on the response code of the login request.

Reply:
An ajax request by default, yes. And as a fallback, the standard post request from the form would redirect via header like it does on a successful login already.